great_expectations/render/renderer/slack_renderer.py

```diff
@@ -1,14 +1,169 @@
+                title_hlink = f"*<{validation_result_urls[0]} | Validation Result>*"
+            else:
+                title_hlink = "*Validation Result*"
+            batch_validation_status_hlinks = "".join(
+                f"*Batch Validation Status*: *<{validation_result_url} | {status}>*"
+                for validation_result_url in validation_result_urls
+            )
+            summary_text += f"""{title_hlink}
+{batch_validation_status_hlinks}
+            """
+        else:
+            summary_text += f"*Batch Validation Status*: {status}"
+
+        summary_text += f"""
+*Expectation Suite name*: `{expectation_suite_name}`
+*Data Asset Name*: `{data_asset_name}`
+*Run ID*: `{run_id}`
+*Batch ID*: `{batch_id}`
+*Summary*: {check_details_text}"""
+
+        if failed_expectations_text:
+            summary_text += failed_expectations_text
+
+        return {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": summary_text,
+            },
+        }
+
+    def _build_report_element_block(
+        self, data_docs_pages: list[dict], notify_with: list[str]
+    ) -> dict | None:
+        if not data_docs_pages:
+            return None
+
+        if notify_with:
+            for docs_link_key in notify_with:
+                if docs_link_key in data_docs_pages.keys():
+                    docs_link = data_docs_pages[docs_link_key]
+                    report_element = self._get_report_element(docs_link)
+                else:
+                    logger.critical(
+                        f"*ERROR*: Slack is trying to provide a link to the following DataDocs: `"
+                        f"{docs_link_key!s}`, but it is not configured under `data_docs_sites` in the "  # noqa: E501
+                        f"`great_expectations.yml`\n"
+                    )
+                    report_element = {
+                        "type": "section",
+                        "text": {
+                            "type": "mrkdwn",
+                            "text": f"*ERROR*: Slack is trying to provide a link to the following DataDocs: "  # noqa: E501
+                            f"`{docs_link_key!s}`, but it is not configured under "
+                            f"`data_docs_sites` in the `great_expectations.yml`\n",
+                        },
+                    }
+                if report_element:
+                    return report_element
+        else:
+            for docs_link_key in data_docs_pages.keys():
+                if docs_link_key == "class":
+                    continue
+                docs_link = data_docs_pages[docs_link_key]
+                report_element = self._get_report_element(docs_link)
+                return report_element
+
+        return None
+
+    def _build_divider_block(self) -> dict:
+        return {"type": "divider"}
+
+    def build_query(self, checkpoint_result: CheckpointResult, text_blocks: list[dict]) -> dict:
```

Comment:
I see the term query used in the old code as well. Reading the code, it looks like this should be `footer`.

Also, the behavior here of this method seems like it could be confusing. mutating text_blocks and then returning another object. I think the most readable solution would just be to have this method just responsible for building the footer section and returning that, then letting `actions.py` do what it needs to with it.